crates/sail-plan/src/resolver/plan.rs

```diff
@@ -2026,6 +2039,146 @@ impl PlanResolver<'_> {
         Ok(plan)
     }
 
+    async fn resolve_query_fill_na(
+        &self,
+        input: spec::QueryPlan,
+        columns: Vec<spec::Identifier>,
+        values: Vec<spec::Expr>,
+        state: &mut PlanResolverState,
+    ) -> PlanResult<LogicalPlan> {
+        if values.len() > 1 && values.len() != columns.len() {
+            return Err(PlanError::invalid(
+                "fill na number of values does not match number of columns",
+            ));
+        }
+
+        let input = self.resolve_query_plan(input, state).await?;
+        let schema = input.schema();
+        let values = self.resolve_expressions(values, schema, state).await?;
+        let columns: Vec<String> = columns.into_iter().map(|x| x.into()).collect();
+
+        let fill_na_exprs = schema
+            .columns()
+            .into_iter()
+            .map(|c| {
+                let column_expr = col(c.clone());
+                let column_data_type = column_expr.get_type(schema)?;
+                let column_name = state.get_field_name(c.name())?;
+                let expr = if columns.is_empty() || columns.iter().any(|col| col == column_name) {
```

Comment:
I guess there may be a way to avoid checking `col == column_name` twice. Let's add a TODO here.